src/ui/Analytics/Analytics.ts

```diff
@@ -254,6 +257,22 @@ export class Analytics extends Component {
       let event = this.componentOptionsModel.getEventName(Model.eventTypes.changeOne + ComponentOptionsModel.attributesEnum.searchHub);
       this.bind.onRootElement(event, (args: IAttributeChangedEventArg) => this.handleSearchHubChanged(args));
     }
+
+    AnalyticsUtils.setClientIdProvider(this);
+  }
+
+  public get visitorId() {
+    return Cookie.get('visitorId');
+  }
+
+  public get clientId() {
+    const clientId = Cookie.get('clientId');
+    if (clientId) {
+      return clientId;
+    }
+    const newclientId = QueryUtils.createGuid();
```

Comment:
`QueryUtils.createGuid` either calls `Crypto` or `Math.random`. The exact two functions used were tested in the collision test (client-side) and no collisions were found after generating more than 2,300,000 UUIDs (Crypto + Math.random).